modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/filters/Filters.tsx

```diff
@@ -134,8 +145,78 @@ function AddFDSFilterModalContent({
 				);
 			}
 		});
+
+		if (filter?.filterType === EFilterType.SELECTION) {
+			const selectionFilter = filter as ISelectionFilter;
+			setSourceType(selectionFilter.sourceType);
+		}
 	}, [filter]);
 
+	const validateForm = ({
+		i18nFilterLabels,
+		isValidDateRange,
+		selectedClientExtension,
+		selectedField,
+		selectedPicklist,
+		sourceType,
+	}: {
+		i18nFilterLabels: Partial<Liferay.Language.FullyLocalizedValue<string>>;
+		isValidDateRange: boolean;
+		selectedClientExtension: IClientExtensionRenderer | undefined;
+		selectedField: IField | undefined;
+		selectedPicklist: IPickList | undefined;
+		sourceType: ESelectionFilterSourceType | undefined;
+	}) => {
+		let isInvalidForm = false;
+
+		if (!selectedField) {
+			isInvalidForm = true;
+		}
```

Comment:
A typical pattern here, that avoids `let` assignment and variable naming woes is:
```js
if (!selectedField) {
    return true;
}

// other if conditions

return false;
```
then, `setSaveButtonDisabled` is invoked outside, based on validation outcome. This improves separation of concerns, as validation logic is isolated from actions based on that validation.

Reply:
yep that makes sense. It's an easy change. next monday we'll start working on the next PR so we can change tthis and move forward